src/Common/src/System/StrongToWeakReference.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System
+{
+    /// <summary>Provides an object wrapper that can transition between strong and weak references to the object.</summary>
+    internal sealed class StrongToWeakReference<T> : WeakReference where T : class
```

Comment:
It's a shame that `WeakReference<T>` is `sealed`.  I wonder if there's a good reason for that?